### PR TITLE
Fix PDF/A validation reporting internal_error after veraPDF 1.30.x upgrade

### LINKED ISSUE #1662: PDF/A validation reports internal_error instead of not_pdfa_compliant after veraPDF upgrade to 1.30.x
## Checked for duplicates
Yes - I've already checked

## 🐛 Describe the bug
When upgrading `org.verapdf:validation-model-jakarta` from 1.28.2 to 1.30.2, the veraPDF library changed behavior in `PDFAParser.getFlavour()`: for PDF files that contain no `pdfaid` conformance declaration in their XMP metadata, veraPDF 1.30+ throws an `IndexOutOfBoundsException` from `GFModelParser.getFlavour()` (accessing an empty list at index 0) instead of returning a default flavour as it did in 1.28.x.

This exception propagates through `PDFUtil.validatePDF()`, gets caught by the outer handler, and is re-thrown as a generic `IOException` with message `"Unable to read with VeraPDF standard reader. Index: 0"`. The caller (`FileReferenceValidationRule.handlePDF`) catches this as `ProblemType.INTERNAL_ERROR` rather than the expected `ProblemType.NON_PDFA_FILE`, causing:

1. Tests expecting `error.pdf.file.not_pdfa_compliant` to fail (the error is reported as `error.validation.internal_error` instead)
2. Tests expecting 0 errors to fail (an unexpected `internal_error` is raised for valid non-PDF/A files)

Affected scenarios from CI: `#1008-1`, `#366-1`, `#824-1`, `#824-2`.

Root cause confirmed by inspection: the test PDFs in these scenarios are plain PDFs exported from Microsoft Word and Ghostscript with no `pdfaid:part` or `pdfaid:conformance` XMP metadata — they are intentionally non-PDF/A and should raise `error.pdf.file.not_pdfa_compliant`.

## 🕵️ Expected behavior
PDF files without a PDF/A conformance declaration should raise `error.pdf.file.not_pdfa_compliant` regardless of veraPDF version. The PDS requirement is that referenced PDF documents must conform to PDF/A-1a (preferred) or PDF/A-1b. A plain PDF with no conformance declaration does not satisfy this requirement and must be flagged as an error.

## 📜 To Reproduce
1. Upgrade `org.verapdf:validation-model-jakarta` from 1.28.2 to 1.30.2 in `pom.xml`
2. Run: `mvn test -Dtest='!ReferenceIntegrityTest*' -Dcucumber.filter.tags='@v3.7.x'`
3. Observe failures in scenarios `#1008-1`, `#366-1`, `#824-1`, `#824-2` with `error.validation.internal_error` instead of `error.pdf.file.not_pdfa_compliant`

## 🖥 Environment Info
- Version of this software: 4.3.0-SNAPSHOT
- veraPDF version: 1.30.2 (regression introduced in 1.30.x)
- Operating System: Linux (GitHub Actions CI)

## 📚 Version of Software Used
- `org.verapdf:validation-model-jakarta:1.30.2`

## 🩺 Test Data / Additional context
Fix: catch `IndexOutOfBoundsException` from `PDFAParser.getFlavour()` in `PDFUtil.validatePDF()` and treat it as a missing conformance declaration (return `false` with an appropriate `errorMessage`) rather than re-throwing as `IOException`. This preserves the correct `NON_PDFA_FILE` error type in the validation report. See PR for fix.

## 🦄 Related requirements
N/A

---

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary

Fixes a regression introduced by upgrading `org.verapdf:validation-model-jakarta` from 1.28.2 to 1.30.2.

**Root cause:** veraPDF 1.30+ changed behavior in `PDFAParser.getFlavour()`. For PDFs with no `pdfaid` conformance declaration in their XMP metadata, the method now throws `IndexOutOfBoundsException` (accessing an empty list at index 0) instead of returning a default flavour as 1.28.x did. The exception propagated through `PDFUtil.validatePDF()`'s outer catch block, was re-thrown as `IOException`, and was then caught by `FileReferenceValidationRule.handlePDF()` as `INTERNAL_ERROR` rather than the expected `NON_PDFA_FILE`.

**Fix:** Catch `IndexOutOfBoundsException` from `getFlavour()` specifically and treat it as a missing conformance declaration — set `errorMessage` and return `false` (non-compliant). Also adds an explicit `NO_FLAVOUR` guard for future veraPDF versions that may return `NO_FLAVOUR` cleanly instead of throwing.

**Confirmed by investigation:**
- All failing test PDFs (`github1008/example.pdf`, `github164/invalid/test1_pdf.pdf`, `github824/1203_12.PDF`) are plain PDFs with no `pdfaid` XMP metadata, verified via `qpdf --qdf --decode-level=none` raw stream extraction
- Under 1.28.2, `getFlavour()` guessed `PDFA_1_B` as a default → validation failed → `NON_PDFA_FILE` raised (correct outcome, accidental path)
- Under 1.30.2, `getFlavour()` throws `IndexOutOfBoundsException` → previously became `INTERNAL_ERROR`

**Changes:**
- `pom.xml`: bump `validation-model-jakarta` 1.28.2 → 1.30.2 (fixes CVE on 1.28.2)
- `PDFUtil.java`: catch `IndexOutOfBoundsException` from `getFlavour()` and treat as missing PDF/A declaration

> 🤖 Generated with [Claude Code](https://claude.ai/claude-code)

**AI Assistance:** This fix was developed with AI assistance (investigation, root cause analysis, and implementation). The logic was verified against actual test PDFs and confirmed passing all targeted test scenarios.

## ⚙️ Test Data and/or Report

Targeted scenarios verified locally with veraPDF 1.30.2:
- `NASA-PDS/validate#1008-1` ✅ `error.pdf.file.not_pdfa_compliant=1`
- `NASA-PDS/validate#366-1` ✅ `error.pdf.file.not_pdfa_compliant=1`
- `NASA-PDS/validate#164-1` ✅ `error.pdf.file.not_pdfa_compliant=1`
- `NASA-PDS/validate#824-1` ✅ 0 errors
- `NASA-PDS/validate#824-2` ✅ 0 errors

Full `@v3.7.x` test suite running in CI.

## ♻️ Related Issues

Fixes #1662
Closes #1625

## 🤓 Reviewer Checklist

- [ ] **Documentation** - Is the new feature/change adequately documented?
- [ ] **Security** - Are there any security implications to these changes?
- [ ] **Testing** - Have tests been added/updated? Are edge cases covered?
- [ ] **Dependencies** - Are dependency changes appropriate and minimal?

### CODE REVIEW

File: src/main/java/gov/nasa/pds/tools/util/PDFUtil.java [hunk 1]
```diff
@@ -89,6 +89,20 @@ private synchronized void writeErrorToFile(String baseDir, String pdfFullName, V
...
+   */
+  private PDFAFlavour detectFlavour(PDFAParser parser) {
+    try {
+      PDFAFlavour flavour = parser.getFlavour();
+      return PDFAFlavour.NO_FLAVOUR.equals(flavour) ? null : flavour;
+    } catch (IndexOutOfBoundsException e) {
+      return null;
+    }
+  }
+
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/tools/util/PDFUtil.java]: You can simplify this and avoid the exception catching by using:
```java
private PDFAFlavour detectFlavour(PDFAParser parser) {
  List<PDFAFlavour> flavours = parser.getFlavours();
  if (flavours.isEmpty()) {
    return null;
  }
  PDFAFlavour flavour = flavours.get(0);
  return PDFAFlavour.NO_FLAVOUR.equals(flavour)? null : flavour;
}
```
Catching `IndexOutOfBoundsException` is a little dodgy here because any `IndexOutOfBoundsException` thrown for any reason inside of `getFlavour` will be interpreted as "This PDF has no conformance declaration".

File: src/main/java/gov/nasa/pds/tools/util/PDFUtil.java [hunk 2]
```diff
@@ -98,35 +112,31 @@ private boolean validatePDF(String baseDir, URI uri, String pdfRef) throws IOExc
         this.errorMessage = "Zero length file is an invalid PDF file.";
         return pdfValidateFlag;
       }
-      // Create a parser and auto-detect flavour
       PDFAParser parser = Foundries.defaultInstance().createParser(new FileInputStream(pdfRef));
-      PDFAFlavour detectedFlavour = parser.getFlavour();
+      PDFAFlavour detectedFlavour = detectFlavour(parser);
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/tools/util/PDFUtil.java:119]: We get ahold of detectedFlavour but then we don't use it on lines 117, 130, 134, and 139. May as well, it's right there; it could neatly replace all those calls to `getFlavour`.

**Review by nutjob4life — APPROVED** [review]
This looks really good although I did point out a couple areas for cleanup. And Maven's tests all pass.

Code delta: ✓
Maven tests: ✓
Approval: ✅
Maven details:
```
303 scenarios (303 passed)
909 steps (909 passed)
4m 56.750s
Aug 24, 2026 6:17:52 PM org.junit.platform.launcher.core.DiscoveryIssueNotifier logIssues
WARNING: TestEngine with ID 'cucumber' encountered a non-critical issue during test discovery:

(1) [WARNING] The classpath resource selector 'features' should not be used to select features in a package. Use the package selector with 'features' instead
[INFO] Tests run: 303, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 296.9 s -- in cucumber.CucumberTest
Aug 24, 2026 6:17:52 PM org.junit.platform.launcher.core.DiscoveryIssueNotifier logIssues
WARNING: TestEngine with ID 'junit-platform-suite' encountered a non-critical issue during test discovery:

(1) [WARNING] [cucumber] The classpath resource selector 'features' should not be used to select features in a package. Use the package selector with 'features' instead (via @Suite cucumber.CucumberTest).
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 323, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
```

File: src/main/java/gov/nasa/pds/tools/util/PDFUtil.java (hunk 1 above)
Comment (jordanpadams) [reply src/main/java/gov/nasa/pds/tools/util/PDFUtil.java]: @copilot can you fix this?

File: src/main/java/gov/nasa/pds/tools/util/PDFUtil.java (hunk 2 above)
Comment (jordanpadams) [reply src/main/java/gov/nasa/pds/tools/util/PDFUtil.java:119]: @copilot can you fix this?

Comment (jordanpadams) [reply src/main/java/gov/nasa/pds/tools/util/PDFUtil.java:119]: @nutjob4life Fixed

File: src/main/java/gov/nasa/pds/tools/util/PDFUtil.java (hunk 1 above)
Comment (jordanpadams) [reply src/main/java/gov/nasa/pds/tools/util/PDFUtil.java]: @nutjob4life fixed
